tests/e2e/system_test/test_graceful_nodes_shutdown.py

```diff
@@ -0,0 +1,736 @@
+            ), f"Huge pages is not applied on {node.name}"
+
+    @pytest.fixture(autouse=True)
+    def setup(
+        self,
+        request,
+        project_factory,
+        pv_encryption_kms_setup_factory,
+        storageclass_factory,
+        pvc_factory,
+        pod_factory,
+        pvc_clone_factory,
+        snapshot_factory,
+        snapshot_restore_factory,
+        mcg_obj,
+        bucket_factory,
+        awscli_pod_session,
+        mcg_obj_session,
+        test_directory_setup,
+    ):
+        """
+        Setting up test requirements
+        """
+        self.amq = AMQ()
+        self.kafka_topic = (
+            self.kafkadrop_pod
+        ) = self.kafkadrop_svc = self.kafkadrop_route = None
+
+        def teardown():
+            logger.info("cleanup the environment")
+
+            if self.amq:
+                if self.kafkadrop_pod:
+                    self.kafkadrop_pod.delete()
+                if self.kafkadrop_svc:
+                    self.kafkadrop_svc.delete()
+                if self.kafkadrop_route:
+                    self.kafkadrop_route.delete()
+
+                self.amq.cleanup()
+
+        request.addfinalizer(teardown)
+
+        logger.info("Starting the test setup")
+
+        # Create a project
+        self.proj_obj = project_factory()
+
+        # Non encrypted fs PVC
+        (
+            self.ne_pvc_obj,
+            self.ne_pvc_pod_obj,
+            self.ne_file_name,
+            self.ne_pvc_orig_md5_sum,
+        ) = self.setup_non_encrypted_pvc(
+            pvc_factory=pvc_factory,
+            pod_factory=pod_factory,
+            pvc_clone_factory=pvc_clone_factory,
+            snapshot_factory=snapshot_factory,
+            snapshot_restore_factory=snapshot_restore_factory,
+        )
+
+        # Encrypted block PVC
+        (
+            self.eb_pvc_obj,
+            self.eb_pvc_pod_obj,
+            self.eb_file_name,
+            self.eb_pvc_orig_md5_sum,
+        ) = self.setup_encrypted_pvc(
+            pv_encryption_kms_setup_factory=pv_encryption_kms_setup_factory,
+            storageclass_factory=storageclass_factory,
+            interface=constants.CEPHBLOCKPOOL,
+            pvc_factory=pvc_factory,
+            pod_factory=pod_factory,
+            pvc_clone_factory=pvc_clone_factory,
+            snapshot_factory=snapshot_factory,
+        )
+
+        # Encrypted fs PVC
+        (
+            self.efs_pvc_obj,
+            self.efs_pvc_pod_obj,
+            self.efs_file_name,
+            self.efs_pvc_orig_md5_sum,
+        ) = self.setup_encrypted_pvc(
+            pvc_factory=pvc_factory,
+            pod_factory=pod_factory,
+            pvc_clone_factory=pvc_clone_factory,
+            snapshot_factory=snapshot_factory,
+        )
+
+        # S3 bucket
+        self.setup_s3_bucket(mcg_obj=mcg_obj, bucket_factory=bucket_factory)
+
+    def setup_amq_kafka_notification(self, bucket_factory):
```

Comment:
see if you can use existing amq common function instead of writing new one here.